odcockpit/src/plugins/environmentviewer/EnvironmentViewerGLWidget.cpp

```diff
@@ -125,13 +125,55 @@ namespace cockpit {
                     m_velodyneSharedMemory(NULL),
                     m_hasAttachedToSharedImageMemory(false),
                     m_velodyneFrame(),
-                    m_CPCreceived(false),
+                    m_12_startingSensorID_32(0),//The first HDL-32E CPC starts from Layer 0
+                    m_11_startingSensorID_32(2),//The second HDL-32E CPC starts from Layer 2
+                    m_9_startingSensorID_32(5),//The third HDL-32E CPC starts from Layer 5
+                    m_12_cpcDistance_32(""),
+                    m_11_cpcDistance_32(""),
+                    m_9_cpcDistance_32(""),
+                    m_previousCPC32TimeStamp(0),
+                    m_cpcMask_32(0),
                     m_cpc(),
                     m_cpcMutex(),
                     m_SPCReceived(false),
                     m_CPCReceived(false),
-                    m_recordingYear(0) {}
-
+                    m_recordingYear(0) {
+                    
+                float sensorIDs_32[32];
```

Comment:
Please use std::array over the old C-style arrays as they are type-safe and provide generic STL-container-like access: http://en.cppreference.com/w/cpp/container/array